File: frontend/src/services/model-config.ts
import request from '@/api/request';

export interface ModelConfigItem {
  id: number;
  provider: string;
  model_name: string;
  temperature: number;
  max_tokens: number;
  rate_limit_per_minute: number;
  daily_quota?: number | null;
  updated_at: string;
}

export interface UpdateModelConfigPayload {
  provider?: string;
  model_name?: string;
  temperature?: number;
  max_tokens?: number;
  rate_limit_per_minute?: number;
  daily_quota?: number | null;
}

export interface ModelUsageLogItem {
  id: number;
  user_id: number;
  model_name: string;
  prompt_tokens: number;
  completion_tokens: number;
  total_tokens: number;
  cost?: number | null;
  created_at: string;
}

export interface ModelUsageLogListResponse {
  items: ModelUsageLogItem[];
  total: number;
}

export interface ListModelUsageLogsParams {
  user_id?: number;
  model_name?: string;
  skip?: number;
  limit?: number;
}

export const getModelConfig = async () => {
  return request.get('/model/config') as Promise<ModelConfigItem>;
};

export const updateModelConfig = async (payload: UpdateModelConfigPayload) => {
  return request.put('/model/config', payload) as Promise<ModelConfigItem>;
};

export const listModelUsageLogs = async (params: ListModelUsageLogsParams = {}) => {
  return request.get('/model/logs', { params }) as Promise<ModelUsageLogListResponse>;
};

File: backend/app/schemas/model_config.py
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, ConfigDict, Field


class ModelConfigRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: int
    provider: str
    model_name: str
    temperature: float
    max_tokens: int
    rate_limit_per_minute: int
    daily_quota: Optional[int]
    updated_at: datetime


class ModelConfigUpdate(BaseModel):
    provider: Optional[str] = None
    model_name: Optional[str] = None
    temperature: Optional[float] = Field(default=None, ge=0, le=2)
    max_tokens: Optional[int] = Field(default=None, gt=0)
    rate_limit_per_minute: Optional[int] = Field(default=None, gt=0)
    daily_quota: Optional[int] = Field(default=None, gt=0)


class ModelUsageLogRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int
    model_name: str
    prompt_tokens: int
    completion_tokens: int
    total_tokens: int
    cost: Optional[float]
    created_at: datetime


class ModelUsageLogListResponse(BaseModel):
    items: List[ModelUsageLogRead]
    total: int
